cmd/capabilities/worker/worker.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+        "log"
+        "fmt"
+        "net/http"
+        "os"
+        "io/ioutil"
+	    "os/exec"
+	    "strings"
+        "github.com/apigee/registry/cmd/capabilities/utils"
+        "encoding/json"
+        "cloud.google.com/go/compute/metadata"
+)
+
+func main() {
+    log.Print("starting server...")
+    http.HandleFunc("/", requestHandler)
+
+    // Determine port for HTTP service.
+    port := os.Getenv("PORT")
+    if port == "" {
+            port = "8080"
+            log.Printf("defaulting to port %s", port)
+    }
+
+    // Start HTTP server.
+    log.Printf("listening on port %s", port)
+    if err := http.ListenAndServe(":"+port, nil); err != nil {
+            log.Fatal(err)
+    }
+}
+
+func getAuthToken() (string, error) {
+    serviceURL := "http://" + os.Getenv("APG_REGISTRY_ADDRESS")
+    tokenURL := fmt.Sprintf("/instance/service-accounts/default/identity?audience=%s", serviceURL)
+    idToken, err := metadata.Get(tokenURL)
+    if err != nil {
+            log.Printf("metadata.Get: failed to query id_token: %+v", err)
+            return "", err
+    }
+
+    return idToken, nil
+}
+
+func requestHandler(w http.ResponseWriter, r *http.Request) {
+    body, err := ioutil.ReadAll(r.Body)
+    if err != nil{
+        log.Printf("ioutil.ReadAll: %v", err)
+        http.Error(w, "Bad Request", http.StatusBadRequest)
```

Comment:
Who is the target audience for this HTTP error message versus the log message above? Can we offer more information about why the request was bad? Bad request indicates the user should do something to fix it, so we should explain where the error is.

Reply:
The workers will receive requests from the Cloud Tasks Queue. So there are no human audiences for the Bad Request response, hence I did not include the details initially. 

However, this makes me realize that providing enough information will be useful for debugging purposes.